src/main/java/com/vou/backend/game/game_info/controller/GameCampaignApiController.java

```diff
@@ -0,0 +1,135 @@
+package com.vou.backend.game.game_info.controller;
+
+import com.vou.backend.game.game_info.dto.GameCampaignRequestDto;
+import com.vou.backend.game.game_info.dto.GameCampaignResponseDto;
+import com.vou.backend.game.game_info.dto.UserCampaignGameResponseDto;
+import com.vou.backend.game.game_info.exception.GameCampaignNotFoundException;
+import com.vou.backend.game.game_info.exception.GameNotFoundException;
+import com.vou.backend.game.game_info.model.GameCampaign;
+import com.vou.backend.game.game_info.model.UserCampaignGame;
+import com.vou.backend.game.game_info.service.GameCampaignService;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * REST controller for managing game campaigns.
+ */
+@RestController
+@RequestMapping("/game-campaigns")
+@RequiredArgsConstructor
+@Validated
+public class GameCampaignApiController {
+    private final GameCampaignService gameCampaignService;
+    private final ModelMapper modelMapper;
+
+    /**
+     * Retrieves all game campaigns.
+     *
+     * @return a list of GameCampaignResponseDto
+     */
+    @GetMapping
+    public ResponseEntity<List<GameCampaignResponseDto>> getAllGameCampaigns() {
+        List<GameCampaign> gameCampaigns = gameCampaignService.findAll();
+        List<GameCampaignResponseDto> responseDtos = gameCampaigns.stream()
```

Comment:
```suggestion
        List<GameCampaignResponseDto> responseDtos = gameCampaigns.stream()
```
Chuyện mapping nên xử lý trong service chứ.

Web layer: chỉ nên xử lý DTO, đồng nghĩa với việc các Controller chỉ nên nhận và trả về dữ liệu là DTO.
Service layer: nhận vào DTO (từ controller gửi qua) hoặc Domain model (từ các service nội bộ khác). Dữ liệu được xử lý (có thể tương tác với DB), cuối cùng được Service trả về Web layer dưới dạng DTO.